src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,42 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+  constructor(name, health) {
+    super(name, health);
+    Animal.alive.push(this);
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+    Animal.alive.push(this);
+  }
+  bite(animal) {
+    if (animal.hasOwnProperty('hidden')) {
+      if (!animal.hidden) {
+        animal.health -= 50;
+      }
+
+      if (animal.health <= 0) {
+        Animal.alive = Animal.alive.filter(
+          (aliveAnimal) => aliveAnimal !== animal,
+        );
+      }
```

Comment:
[CODE PERFORMANCE] - The current implementation of removing the dead animal from the `Animal.alive` array is correct, but be cautious about the performance implications if the array becomes very large. The `filter` method creates a new array and could be less efficient than other methods, such as using `splice` in combination with an index. However, as per the checklist, using `filter` is the preferred approach, so there is no need to change this unless performance becomes an issue.